packages/messaging/lib/messages/PayoutCurvePiece.ts

```diff
@@ -0,0 +1,225 @@
+import { BufferReader, BufferWriter } from '@node-lightning/bufio';
+import { MessageType } from '../MessageType';
+import { IDlcMessage } from './DlcMessage';
+
+export abstract class PayoutCurvePiece {
+  public static deserialize(
+    buf: Buffer,
+  ): PolynomialPayoutCurvePiece | HyperbolaPayoutCurvePiece {
+    const reader = new BufferReader(buf);
+
+    const type = Number(reader.readBigSize());
+
+    switch (type) {
+      case MessageType.PolynomialPayoutCurvePiece:
+        return PolynomialPayoutCurvePiece.deserialize(buf);
+      case MessageType.HyperbolaPayoutCurvePiece:
```

Comment:
Should this have a version number? Not sure what the answer is so added this to DLC Spec: https://github.com/discreetlogcontracts/dlcspecs/pull/144/files#r599245598

Reply:
yeah, isn’t specified in spec, so added a temp one for now